File: CustomSoft.BackEnd/Application/ViewModels/Book/BookViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.ViewModels.Book
{
    public class BookViewModel
    {
        public BookViewModel()
        {

        }
        public  Guid BookId { get; set; }
        public  string Title { get; set; }
        public string? FileDirection { get; set; }
        public string? FileName { get; set; }
        public string? PublicationDate { get; set; }
        public string BookAuthorGuid { get; set; }
        public string AuthorName { get; set; }
        public string AuthorLastName { get; set; }
        public string? Birthdate { get; set; } 
        public string? ErrorMsj { get; set; } 
    }
}

File: CustomSoft.Front/viewModel/BookViewModel.ts
export interface Book {
    // Define las propiedades de acuerdo a tu modelo de datos
    BookId: string;
    Title: string;
    FileDirection?: string;
    PublicationDate?: Date;
    BookAuthorGuid?: string;
    AuthorName?: string;
    AuthorLastName?: string;
    AuthorBirthdate?: string;
    FileName?: string;
}


export class BookViewModel implements Book {
    BookId: string = '';

    Title: string = '';
    PublicationDate?: Date;
    BookAuthorGuid?: string = '';
    AuthorName?: string = '';
    AuthorLastName?: string = '';
    FileName?: string = '';

    constructor(
        BookId: string,
        Title: string,
        PublicationDate?: Date,
        BookAuthorGuid?: string,
        AuthorName?: string,
        AuthorLastName?: string,
        FileName?: string
    ) {
        this.BookId = BookId;
        this.Title = Title;
        this.PublicationDate = PublicationDate;
        this.BookAuthorGuid = BookAuthorGuid;
        this.AuthorName = AuthorName;
        this.AuthorLastName = AuthorLastName;
        this.FileName = FileName;
    }

    static createEmpty(): BookViewModel {
        return new BookViewModel('','');
    }


}
